src/write_ext.rs

```diff
@@ -0,0 +1,108 @@
+use crate::consts::*;
+use byteorder::{LittleEndian, WriteBytesExt};
+
+/// Extension traits for io::Read and io::Write to read and
+/// write bin_prot encoded types
+use std::io;
+
+// extension trait for writers implementing io::Write to allow them to write
+// the primitive values for bin_prot
+pub trait WriteBinProtExt: io::Write {
+    fn bin_write_unit(&mut self) -> Result<usize, io::Error> {
+        self.write_u8(0x00).map(|_| 1)
+    }
+
+    fn bin_write_bool(&mut self, b: bool) -> Result<usize, io::Error> {
+        self.write_u8(if b { 0x01 } else { 0x00 }).map(|_| 1)
+    }
+
+    // chars are utf-8 so can be 1-4 bytes long
+    fn bin_write_char(&mut self, c: char) -> Result<usize, io::Error> {
```

Comment:
is c not being used? The method writes null bytes and does not convert `c` to utf-8 bytes. Probably need [encode_utf8](https://doc.rust-lang.org/std/primitive.char.html#method.encode_utf8)